server/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java

```diff
@@ -220,160 +209,133 @@ public void beforeIndexShardClosed(ShardId shardId, @Nullable IndexShard indexSh
      * @param event cluster state changed event
      */
     private void processIndexShardSnapshots(ClusterChangedEvent event) {
```

Comment:
perhaps only pass current cluster state to this method

Reply:
Yea, we can even just pass `SnapshotsInProgress` since we have that in the caller already :) Will simplify.